.github/workflows/ci.yml

```diff
@@ -29,39 +29,42 @@ jobs:
         uses: actions/cache@v2
         with:
           path: target/
-          key: ${{ matrix.config.os }}-${{ hashFiles('Cargo.lock') }}
+          key: ${{ matrix.os }}-${{ hashFiles('Cargo.lock') }}
 
       - name: Environment
         run: |
           echo ::set-env name=GH_ACTIONS::1
           echo ::set-env name=RUST_BACKTRACE::full
 
-      - name: lint
-        run: cargo clippy --all-targets --locked -- -D clippy::all
+      - name: Build
+        run: cargo build --locked --release --all-targets
 
-      - name: format
-        run: rustfmt --check src/main.rs
+      - name: Test
+        run: cargo test --locked --release --all-targets
+
+      - name: Lint
+        run: cargo clippy --all-targets --release --locked -- -D clippy::all
 
-      - name: test
-        run: cargo test --locked --all-targets
+      - name: Format
```

Comment:
Can't we do this before build, test and lint so the CI fails nice and quick if fmt is off?

Reply:
Sure